pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -209,13 +211,44 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 	pod.Spec.Volumes = append(pod.Spec.Volumes, podVolumes...)
 
 	// Inject credentials
-	if err := mi.credentialBuilder.CreateSecretVolumeAndEnv(
-		pod.Namespace,
-		pod.Spec.ServiceAccountName,
-		initContainer,
-		&pod.Spec.Volumes,
-	); err != nil {
-		return err
+	hasStorageSpec, ok := pod.ObjectMeta.Annotations[constants.StorageSpecAnnotationKey]
+	storageKey, storageKey_exist := pod.ObjectMeta.Annotations[constants.StorageSpecKeyAnnotationKey]
+	if !storageKey_exist {
+		storageKey = mi.config.DefaultCredentialKey
```

Comment:
Actually our thinking was to default this to a key name based on the type, for example if s3 it would be `s3_default`, etc. So probably better to just leave the key empty here (no need for the separate _exist var) and have the logic in `CreateStorageSpecSecretVolumeAndEnv` sort that out.